pi5/board.go

```diff
@@ -159,19 +149,109 @@ func (b *pinctrlpi5) Reconfigure(
 		return err
 	}
 
-	b.mu.Lock()
-	defer b.mu.Unlock()
+	// make sure every pin has a name. We already know every pin has a pin
+	for _, c := range newConf.Pins {
+		if c.Name == "" {
+			c.Name = c.Pin
+		}
+	}
+
+	if err := b.addUserDefinedNames(newConf); err != nil {
+		return err
+	}
 
 	if err := b.reconfigurePullUpPullDowns(newConf); err != nil {
 		return err
 	}
+	if err := b.reconfigureInterrupts(newConf); err != nil {
+		return err
+	}
+	b.pinConfigs = newConf.Pins
```

Comment:
I'd think you'd want to lock for this, but I see it was deleted

Reply:
actually I'd think for this whole function, what's the reasoning for removing the lock?

Reply:
So I needed to remove the lock in order to call `DigitalInterruptByName` within `reconfigureInterrupts`

I didn't think anything else would be trying to access the board while reconfigure was in progress, but I suppose that might not be true. 

I'll move the important parts of `DigitalInterruptByName` into a helper function that can be called within reconfigure

Reply:
added back the lock